docs/Components-TextField.md

```diff
@@ -0,0 +1,146 @@
+# TextField Component
+
+[Back to all components](./Components.md)
+
+Ignite's `TextField` Component is an enhanced version of the built-in React Native [`TextInput`](https://reactnative.dev/docs/textinput) component. It consists of TextInput and a Text(./Components-Text.md) label. 
+
+With this component you will be able to standardise TextInput component across your app.
+
+```tsx
+import { TextField } from '../components';
+
+const [input, setInput] = useState("")
+<TextField
+  value={input}
+  onChangeText={(value) => setInput(value)}
+  label="Name"
+  placeholder="omg your name"
+  style={$header}
+  inputStyle={$inputStyle}
+  preset="default"
+/>
+
+```
+
+## Props
+
+### `label`
+
+The `label` prop is optional. We encourage you to not use this but rather use the `labelTx` prop instead.
+
+```tsx
+<TextField
+  value={input}
+  onChangeText={(value) => setInput(value)}
+  label="Name"
+/>
+```
+
+### `labelTx`
+
+The `labelTx` prop is optional. The label text is looked up by [i18n](https://www.i18next.com/) (used for internationalization)
```

Comment:
I think the library we use is `i18n-js`, so I'd probably link to that, and maybe link to the `Component-Text.md` documentation, as these props are passed to that component.